### PR TITLE
New paths may seed initial_rtt

### LINKED ISSUE #2644: RTT measurement using PATH_* frames
Should the recovery draft explain how to gather RTT samples during connection migration using PATH_CHALLENGE and PATH_RESPONSE?

This is explained in:
- https://tools.ietf.org/html/draft-paulo-quic-migration-00#section-3.3

### PR DESCRIPTION
**Pull request by ianswett**
Using PATH_CHALLENGE/PATH_RESPONSE

I'm fairly happy with the text, but not that happy with the location.  Especially after the recently filed issues(ie: #2684) on resetting the congestion controller on path changes, I think we may need a section in recovery on that.

Fixes #2644

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -521,6 +521,10 @@ is available, or if the network changes, the initial RTT SHOULD be set to 500ms,
 resulting in a 1 second initial handshake timeout as recommended in
 {{?RFC6298}}.
 
+A new path on an existing connection MAY use the delay between sending a
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: can you fix this sentence? the path does not use the delay

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Better?

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -521,6 +521,10 @@ is available, or if the network changes, the initial RTT SHOULD be set to 500ms,
 resulting in a 1 second initial handshake timeout as recommended in
 {{?RFC6298}}.
 
+A connection MAY use the delay between sending a PATH_CHALLENGE and receiving
+a PATH_RESPONSE to seed initial_rtt for a new path, but MUST not be considered
```
Comment (dtikhonov) [inline draft-ietf-quic-recovery.md]: Consider adding "the delay" -- "..., but _the delay_ MUST not be considered..."

In the current form, the sentence suggests that the connection must not be considered an RTT sample.

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Thanks, done.

**Comment by martinthomson** [comment]
1. Do we really want to keep the requirement to respond with a PATH_RESPONSE *immediately*?

2. This doesn't talk about paths.  Now, we might infer from this that the RTT to a peer isn't likely to dramatically change when paths change, but we'd want to be saying that directly.

3. How does this sit with the change in #2637?

**Comment by mikkelfj** [comment]
On 2.: if you roam from 5G to 3G you could see significant latency spikes. Phasing out 3G networks and keeping 2G could move traffic to 2G EDGE in areas of thin coverage. Traffic could also move to narrow band IoT (if it is possible to run QUIC on these at all) with MUCH higher latency. Endpoints could choose stay on narrow band most of the time to save battery and perhaps online costs, but use 5G when a device wakes up for real (I could see a use case for that). Though of course might be served with separate connections for such extreme transitions.

**Comment by ianswett** [comment]
1. I think this is OK as long as we limit how many path responses we send immediately?  Realistically, it's not that different from a generate_204, but it's something to be aware of.
2. I think we need to add a whole section on paths and connection migration in recovery. See #2684   I can do that in this PR if people want? 
3. #2367 decreases the value of this, since you may only see one half of the path.  It might still be better than nothing?  Initial RTT doesn't need to be correct, but getting it within a factor of 2 is helpful.

**Comment by DavidSchinazi** [comment]
@ianswett I don't think you meant 2367

**Comment by ianswett** [comment]
Thanks David, fixed.

**Comment by janaiyengar** [comment]
I'm split on this. While I understand the urge to optimize here, I'm not sure that documenting it here is necessary. 

Saying that an endpoint MAY use this signal gives permission, but does so without noting the potential issues. At a minimum, you would need to talk about the problems that are lurking ... especially in light of #2637, as I've noted in [my comment on #2644](https://github.com/quicwg/base-drafts/issues/2644#issuecomment-490721850).

Which leads me to wondering if it's really worth documenting this here without more experimentation and data.

**Comment by ianswett** [comment]
I'm a fan of forbidding it's use as an actual RTT measurement, which would be one reason to say something?

**Comment by rpaulo** [comment]
I'd prefer that it's NOT forbidden.  I'm fine with waiting for experimentation.

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -521,6 +521,10 @@ is available, or if the network changes, the initial RTT SHOULD be set to 500ms,
 resulting in a 1 second initial handshake timeout as recommended in
 {{?RFC6298}}.
 
+A connection MAY use the delay between sending a PATH_CHALLENGE and receiving
+a PATH_RESPONSE to seed initial_rtt for a new path, but the delay MUST not be
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
a PATH_RESPONSE to seed initial_rtt for a new path, but the delay MUST NOT be
```

**Comment by ianswett** [comment]
I changed this to SHOULD NOT from MUST NOT to allow some experimentation.

**Pull request merged**